core/views.py

```diff
@@ -1,32 +1,61 @@
+from django.views.generic.base import TemplateView
 import stripe
 
 from django.conf import settings
 from django.shortcuts import render
 from django.views.generic import View
+from django.contrib.auth.views import LogoutView, LoginView
+from django_registration.backends.one_step.views import RegistrationView
 
-from subscription.models import Customer, Product
+from subscription.models import Customer
 
 stripe.api_key = settings.STRIPE_SECRET_KEY
 
 
-class HomeView(View):
-    def get(self, request):
-        if not request.user.is_authenticated:
-            return render(request, 'core/home.html')
-        else:
+class Login(LoginView):
+    template_name = "core/registration/login.html"
+
+
+class Logout(LogoutView):
+    template_name = "core/registration/logout.html"
+
+
+class RegisterView(RegistrationView):
+    template_name = "core/django_registration/signup.html"
+    success_url = "complete"
+
+
+class RegistrationCompleteView(RegistrationView):
+    template_name = "core/django_registration/signup_complete.html"
+
+
+class HomeView(TemplateView):
+    template_name = "core/home.html"
+
+    def get_context_data(self, *args, **kwargs):
+        context = super(HomeView, self).get_context_data(*args, **kwargs)
+
+        if self.request.user.is_authenticated:
```

Comment:
Does this check even matters? I don't see any separate authentication based step below! Remove this!